packages/node_modules/@ciscospark/plugin-phone/src/state-parsers.js

```diff
@@ -70,10 +70,85 @@ export function joinedOnThisDevice(spark, locus) {
 }
 
 /**
+ * Converts a list of participants to a list of memberships
+ * @param {SparkCore} spark
+ * @param {Types~Locus} locus
+ * @private
+ * @returns {Array<CallMembership>}
+ */
+export function participantsToCallMemberships(spark, locus) {
+  const users = new Map();
+  const devices = new Map();
+
+  for (const participant of locus.participants) {
+    if (participant.type === `USER`) {
+      users.set(participant.url, participant);
+    }
+    else if (participant.type === `RESOURCE_ROOM`) {
+      devices.set(participant.url, participant);
+    }
```

Comment:
Do we want to track unknown participants?

Reply:
no one has been able to tell me what `UNKNOWN` means, so, at this time, `USER` and `RESOURCE_ROOM` seem to be the only types that provide useful information. The fourth type, `MEETING_BRIDGE` has been officially established to provide no client-useful information (though I can imagine a future where it provides UI-relevant information to call but not any specific member)

Reply:
Gotcha.